helix-core/src/main/java/org/apache/helix/controller/stages/ManagementModeStage.java

```diff
@@ -146,6 +145,13 @@ private void recordClusterStatus(ClusterManagementMode mode, HelixDataAccessor a
     ClusterManagementMode.Type recordedType = clusterStatus.getManagementMode();
     ClusterManagementMode.Status recordedStatus = clusterStatus.getManagementModeStatus();
 
+    if (mode.getMode().equals(recordedType) && mode.getStatus().equals(recordedStatus)) {
```

Comment:
This is more like a patch for history. If there are duplicates, it suppose to be something in logic error.

Reply:
No. It's because of the timing. Ex: 2 live instance change events A and B. For A, the cache refresh already gets all the changes and calculates that the status is fully frozen. And event B, the result is the same.
So I would say, the duplicate status result is expected. But we don't want to write it again to reduce the writes.

Reply:
If you already refreshed and get the status, then you already know what status is. So the history recording is idempotent.  There should not be any duplication. I would suggest to have an offline sync.